reader_permit.hh

```diff
@@ -276,7 +276,12 @@ public:
 
     T* allocate(size_t n) {
         auto p = _alloc.allocate(n);
-        _permit.consume(reader_resources::with_memory(n * sizeof(T)));
+        try {
+            _permit.consume(reader_resources::with_memory(n * sizeof(T)));
+        } catch (...) {
+            _alloc.deallocate(p, n);
+            throw;
+        }
```

Comment:
This looks real. But missing Fixes #?

Reply:
Right, will need to backport to 5.2.
I will create an issue.

Reply:
Created issue and added it to cover letter.